### ISSUE TITLE
Reach

### ISSUE THREAD

**Issue by bwhetherington** [bwhetherington/dnd-5-5e#6]
There are a number of effects throughout the game which require a target be within 5 feet of you to trigger attacks, such as Berserker's Retaliation feature. Additionally, spells such as Green-Flame Blade and Booming Blade have the concept of 5 feet of reach built into them.

I think that these features should instead be reworded to be based on the reach of whatever weapon you use for the attack, and that such language be used more broadly throughout. A creature's reach could be a standardized term that we use in place of the current 5 feet range.

**Comment by JamesMcClung** [comment]
I agree that Retaliation and similar martial features should be reach-based. I think something like
> #### Retaliation
> When you take damage from a creature, you can use your reaction to make a melee weapon attack against that creature if it is within your reach.
is acceptable even within the current rules, although it does evoke questions such as "what if you can't see the creature?" and "what if it's behind cover?".

I suspect that BB was deliberately hardcoded to 5 feet so as to prevent tactics involving BBlade from outside of your opponent's reach, forcing them to approach you to hit you. GFB would have been changed just to preserve symmetry.[^1] But again, the phrasing "make a melee weapon attack against a creature within reach" is plainly understandable, I think. That, and making the range simply "self" would fix the two cantrips.

Anyways, I agree that "reach" should probably be used more commonly. I'm not sure how you would redefine it, though (it's defined on PH 195, which also exhibits the phrasing I used above: "a melee attack allows you to attack a foe within your reach").

[^1]: This tactic isn't even effective. To actually be good, one needs a way to BBlade first and then run away without provoking. Cunning Action and Mobile make this legitimate tactic pretty accessible, and it is independent of reach.

**Comment by JamesMcClung** [comment]
One benefit of reworking the mechanics of reach could address jank builds that extend reach using multiple sources; there was some bugbear build that could get a 20 ft reach, for instance. Preventing reach stacking or using phrasing such as "your reach is extended 5 feet _beyond your weapon's usual reach_" would be good.

**Comment by bwhetherington** [comment]
With such standardized language, it could then be reasonable to add reach bonuses into places like, say, the Enlarge spell, and Rune Knight's Giant's Might, as long we are are no longer worried about stacking such bonuses arbitrarily.

**Comment by JamesMcClung** [comment]
Agreed
